src/library.py

```diff
@@ -495,3 +495,236 @@ def check_detail(self) -> Dict:
         print("Verification results dict: ")
         print(output)
         return output
+
+
+class AutomaticOADamperControl(CheckLibBase):
+    points = ["no_of_occ", "m_oa", "eco_onoff"]
+
+    def verify(self):
+        tol = 0.03
+        self.result = ~(
+            (self.df["no_of_occ"] < tol)
+            & (self.df["m_oa"] > 0)
+            & (self.df["eco_onoff"] == 0)
+        )
+
+    def check_bool(self) -> bool:
+        if len(self.result[self.result == True] > 0):
+            return True
+        else:
+            return False
+
+    def check_detail(self):
+        print("Verification results dict: ")
+        output = {
+            "Sample #": len(self.result),
+            "Pass #": len(self.result[self.result == True]),
+            "Fail #": len(self.result[self.result == False]),
+            "Verification Passed?": self.check_bool(),
+        }
+        print(output)
+        return output
+
+
+class FanStaticPressureResetControl(CheckLibBase):
+    points = ["p_set", "d_VAV_1", "d_VAV_2", "d_VAV_3", "d_VAV_4", "d_VAV_5"]
+
+    def verify(self):
+        tol = 1
+        d_vav_points = ["d_VAV_1", "d_VAV_2", "d_VAV_3", "d_VAV_4", "d_VAV_5"]
+        d_vav_df = self.df[d_vav_points].copy(deep=True)
+        self.df["result"] = True
+        self.df.at[
+            0, "result"
+        ] = True  # set first row True, reason: first row can't be determined
+
+        for row_num, (index, row) in enumerate(self.df.iterrows()):
+            if row_num != 0:
+                if (
+                    self.df.at[index, "p_set"] > self.df.at[prev_index, "p_set"] + tol
+                    and (d_vav_df.iloc[row_num] < 0.9).all()
+                ):
+                    self.df.at[index, "result"] = False
+            prev_index = index
+
+        self.result = self.df["result"].copy(deep=True)
+
+    def check_bool(self) -> bool:
+        if len(self.result[self.result == True] > 0):
+            return True
+        else:
+            return False
+
+    def check_detail(self):
+        print("Verification results dict: ")
+        output = {
+            "Sample #": len(self.result),
+            "Pass #": len(self.result[self.result == True]),
+            "Fail #": len(self.result[self.result == False]),
+            "Verification Passed?": self.check_bool(),
+        }
+        print(output)
+        return output
+
+
+class HeatRejectionFanVariableFlowControlsCells(CheckLibBase):
+    points = [
+        "ct_op_cells",
+        "ct_cells",
+        "m",
+        "P_fan_ct",
+        "m_des",
+        "min_flow_frac_per_cell",
+    ]
+
+    def verify(self):
+        self.df["ct_cells_op_theo"] = np.nan
+        for index, row in self.df.iterrows():
+            self.df.at[index, "ct_cells_op_theo"] = min(
+                int(
+                    (
+                        self.df.loc[index, "m"]
```

Comment:
I believe ```index``` is required according to this documentation [link](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.loc.html#:~:text=df.loc%5B%27cobra%27%2C%20%27shield%27%5D).  Also, I implemented the code without ```index``` but, I got an error because of not having an ```index```. It might be because ```index``` is date instead of integer value?

Reply:
does row["m"] work and give you the same value for LOC 586?

Reply:
```row["m"]``` (without ```index```) caused an error. Do you usually not specify ```index``` when using ```.loc``` method?

Reply:
when you already have row, you should not need to do .loc, when using .loc, yes you will need to specify index.

Reply:
Fixed.